base_optimization.php

```diff
@@ -0,0 +1,284 @@
+<?php
+
+abstract class base_optimization
```

Comment:
I suggest to use PascalCase name for abstract class rather than using underscores.